owltorrent-network/src/main/java/edu/rice/owltorrent/network/SocketConnector.java

```diff
@@ -0,0 +1,136 @@
+package edu.rice.owltorrent.network;
+
+import edu.rice.owltorrent.common.adapters.NetworkToStorageAdapter;
+import edu.rice.owltorrent.common.entity.Peer;
+import edu.rice.owltorrent.network.messagereader.BusyWaitMessageReader;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Socket implementation of PeerConnector class.
+ *
+ * @author Lorraine Lyu, Max Yu
+ */
+@Log4j2(topic = "general")
+public class SocketConnector extends PeerConnector {
+  private final Socket peerSocket;
+  private DataOutputStream out;
+  private DataInputStream in;
+
+  private final Runnable listenForInput =
+      new Runnable() {
+        @Override
+        public void run() {
+          ReadableByteChannel channel = Channels.newChannel(in);
+          while (true) {
+            try {
+              messageReader.readMessage(channel);
+            } catch (IOException e) {
+              log.error(e);
+            }
+          }
+        }
+      };
+
+  private SocketConnector(
+      Peer peer,
+      NetworkToStorageAdapter storageAdapter,
+      MessageReader messageReader,
+      Socket peerSocket) {
+    super(peer, storageAdapter, messageReader);
+    this.peerSocket = peerSocket;
+  }
+
+  public static SocketConnector initiateConnection(
+      Peer peer, NetworkToStorageAdapter storageAdapter) throws IOException {
+    SocketConnector connector =
+        new SocketConnector(
+            peer,
+            storageAdapter,
+            new BusyWaitMessageReader(),
+            new Socket(peer.getAddress().getAddress(), peer.getAddress().getPort()));
+    connector.initiateConnection();
+    return connector;
+  }
+
+  public static SocketConnector respondToConnection(
+      Peer peer, Socket peerSocket, NetworkToStorageAdapter storageAdapter) throws IOException {
+    SocketConnector connector =
+        new SocketConnector(peer, storageAdapter, new BusyWaitMessageReader(), peerSocket);
+    connector.respondToConnection();
```

Comment:
This should be called something else from the static method

Reply:
Not sure what to rename it to so kept it for now. (Also to avoid messy merges)

Reply:
Not a good reason not to change, but for the sake of keeping development moving I'll allow it. At least would be nice to add a TODO in the code.